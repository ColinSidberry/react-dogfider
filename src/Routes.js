import { Switch, Route, Redirect } from "react-router-dom";
import DogDetails from "./DogDetails";
import DogList from "./DogList";


function Routes({ dogs }) {
    // console.log("route dogs", dogs) //Works
    return (
        <div>
            <Switch>
                <Route exact path="/dogs">
                    <DogList dogs={dogs} />
                </Route>
                <Route exact path="/dogs/:name">
                    <DogDetails dogs={dogs} />
                </Route>
                <Redirect to="/dogs" />
            </Switch>

        </div>
    )
}

export default Routes;