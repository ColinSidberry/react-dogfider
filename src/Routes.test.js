import { MemoryRouter } from "react-router-dom";
import Routes from "./Routes";
import { render } from "@testing-library/react";

const sampleDogs = [
    {
        name: "testDog1",
        src: "testDog1",
        age: 2,
        facts: ["barks", "eats"]
    },
    {
        name: "testDog2",
        src: "testDog2",
        age: 2,
        facts: ["plays", "barks"]
    }]

it("renders without crashing", function () {
    render(
        <MemoryRouter>
            <Routes dogs={sampleDogs} />
        </MemoryRouter>
    )
})

it("matches snapshot", function () {
    const { container } = render(
        <MemoryRouter>
            <Routes dogs={sampleDogs} />
        </MemoryRouter>
    )
    expect(container).toMatchSnapshot();
})

it("can show all dogs", function () {
    const { getByText } = render(
        <MemoryRouter initialEntries={["/dogs"]}>
            <Routes dogs={sampleDogs} />
        </MemoryRouter>
    )
    const titleText = getByText(/testDog1/i);  // QUESTION: what is this slash thing?????
    // debug(titleText);
    const titleText2 = getByText(/testDog2/i);

    expect(titleText).toBeInTheDocument();
    expect(titleText2).toBeInTheDocument();

})


it("can show an idividual dog", function () {
    const { container, getByText } = render(
        <MemoryRouter initialEntries={["/dogs/testdog1"]}>
            <Routes dogs={sampleDogs} />
        </MemoryRouter>
    )
    const titleText = getByText(/testDog1/i);  // QUESTION: what is this slash thing?????
    // debug(titleText);
    const ageText = getByText(/age: 2/i);

    // debug(container)
    const lists = container.querySelectorAll("li")

    expect(titleText).toBeInTheDocument();
    expect(ageText).toBeInTheDocument();
    expect(lists).toHaveLength(2);

})