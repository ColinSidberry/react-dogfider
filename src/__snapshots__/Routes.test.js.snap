// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div>
    <div>
      <div
        class="ShowDogDetail"
      >
        <h1>
          <a
            href="/dogs/testDog1"
          >
            testDog1
          </a>
           
        </h1>
        <img
          alt="testDog1"
          class="ShowDogDetail-image"
          src="/testDog1.jpg"
        />
        <h2>
           Age: 
          2
        </h2>
        <ul>
          <li>
            barks
          </li>
          <li>
            eats
          </li>
        </ul>
      </div>
      <div
        class="ShowDogDetail"
      >
        <h1>
          <a
            href="/dogs/testDog2"
          >
            testDog2
          </a>
           
        </h1>
        <img
          alt="testDog2"
          class="ShowDogDetail-image"
          src="/testDog2.jpg"
        />
        <h2>
           Age: 
          2
        </h2>
        <ul>
          <li>
            plays
          </li>
          <li>
            barks
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
